test/e2e/network/kube_proxy.go

```diff
@@ -230,11 +231,14 @@ var _ = SIGDescribe("Network", func() {
 			string(jsonBytes))
 		// Run the closewait command in a subroutine so it keeps waiting during postFinTimeoutSeconds
 		// otherwise the pod is deleted and the connection is closed loosing the conntrack entry
-		go func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		go func(ctx context.Context) {
 			defer ginkgo.GinkgoRecover()
-			_, err = framework.RunHostCmd(fr.Namespace.Name, "e2e-net-client", cmd)
-			framework.ExpectNoError(err)
-		}()
+			// swallow the error here because the test will fail if the command fails
+			// but it will report an error during the cleanup of the test
+			framework.RunHostCmd(fr.Namespace.Name, "e2e-net-client", cmd)
```

Comment:
the context is not propagated to RunHostCmd... how does it help with cleanup?

Reply:
🤦‍♂ I was thinking that it will close the goroutine, you are right, it doesn't help at all 
Would love any suggestions, is the only option open a channel and wait until this goroutine ends?

Reply:
what exactly is causing the problem? is it the goroutine, or the running kubectl exec command, or something else?

Reply:
this command run in a pod in parallel and takes `postFinTimeoutSeconds = 30 sec`  to finish.
Then, the test tries to find the conntrack entry, and it usually finds it in less than 5 secs, and finish the test deleting the pod where the command is being executed.

This deletion of the pod does that this command throws an error code 137 ,that causes the subsequent test to fail, because ginkgo assert the error in the next test that is executed
```
_, err = framework.RunHostCmd(fr.Namespace.Name, "e2e-net-client", cmd)		
			framework.ExpectNoError(err)
```
I thought that with this I could cancel the goroutine when the test finish, but is clear that is not the case, and seems the reason why it works now is because the errors is swallowed